rust/src/measurements/toeplitz/utils/mod.rs

```diff
@@ -0,0 +1,4 @@
+pub(crate) mod core;
+pub(crate) mod isotonic;
+pub(crate) mod noise_generation;
+pub(crate) mod type_conversion;
```

Comment:
It looks a bit odd to have a whole `utils` module that actually contains the core functionality as a submodule, since `core.rs` has the most important code imo. I would be fine with a flat `toeplitz` module containing all the submodules.